### ISSUE TITLE
False positive by lines-around-comment for JSDoc

### ISSUE THREAD

**Issue by revolter** [eslint/eslint#10671]
**Tell us about your environment**

* **ESLint Version:** 5.2.0
* **Node Version:** v8.9.4
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser": true,
        "es6": true,
        "greasemonkey": true
    },
    "extends": "eslint:recommended",
    "globals": {
        "min": false
    },
    "rules": {
        "lines-around-comment": [
            "error", {
                "afterBlockComment": true
            }
        ]
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**
 * Converts an array-like object to an Array.
 * @param {Object} collection - The collection to be converted.
 * @returns {Object[]} The resulting array.
 */
function toArray(collection) {
    return Array.prototype.slice.call(collection);
}

toArray([]);
```

```bash
eslint scripts/**/*
```

**What did you expect to happen?**

Not throw an error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
...scripts/Main/main.js
  1:1  error  Expected line after comment  lines-around-comment

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

I think that it should make an exception for JSDoc. Meaning that `/** */` is not actually a block comment, `/* * */` is a block comment.

The current workaround is to use `"ignorePattern": "\\*"`.

**Comment by platinumazure** [comment]
Hi @revolter, thanks for the issue.

Are JSDoc blocks required to be next to the code element, according to the spec? If so, I definitely see this as a bug.

Otherwise, if JSDoc blocks don't have to be immediately above their matching code element, I think this could become an enhancement request for an option to support JSDoc blocks not having a line after them.

Any thoughts? (Also curious if anyone from @eslint/eslint-team has any thoughts.)

**Comment by kaicataldo** [comment]
It looks like we get around this in our codebase with the following config: https://github.com/eslint/eslint/blob/master/packages/eslint-config-eslint/default.yml#L35-L40

**Comment by revolter** [comment]
This is still an issue.

**Comment by kaicataldo** [comment]
Given that [JSDoc support has been deprecated](https://eslint.org/blog/2018/11/jsdoc-end-of-life), JSDoc comments won’t have any special meaning to ESLint core rules going forward. If `ignorePattern` doesn’t solve the issue reported here, do you mind making a new issue proposing an enhancement that might?
